src/components/sbb-wagon/sbb-wagon.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Element, Fragment, h, JSX, Prop, State } from '@stencil/core';
+import { InterfaceSbbWagonAttributes } from './sbb-wagon.custom.d';
+
+let nextId = 0;
+
+/**
+ * @slot unnamed - Used to slot one to x icons for meta information of the sbb-wagon.
+ */
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-wagon.scss',
+  tag: 'sbb-wagon',
+})
+export class SbbWagon {
+  /** Wagon type. */
+  @Prop({ reflect: true }) public type: InterfaceSbbWagonAttributes['type'] = 'wagon';
+
+  /** Occupation icon of a wagon. */
+  @Prop() public occupancy: InterfaceSbbWagonAttributes['occupancy'] = 'none';
+
+  /** Visible class label of a wagon. */
+  @Prop() public wagonClass?: '1' | '2';
+
+  /** Visible label for the wagon number. Not used by type locomotive or blocked. */
+  @Prop() public label?: string;
+
+  /** Accessibility text for translations to describe the wagon type. */
+  @Prop() public accessibilityLabelWagon = '';
+
+  /** Accessibility text for translations to add additional information to wagon. */
+  @Prop() public accessibilityAdditionalWagonText = '';
+
+  /** Accessibility text for translations to describe the occupation level of a wagon. */
+  @Prop() public accessibilityLabelOccupation = '';
+
+  /** Accessibility text for translations to describe the class of a wagon. */
+  @Prop() public accessibilityLabelClass = '';
+
+  /** Accessibility-text for translations as the list title for additional information icons on a wagon. */
+  @Prop() public accessibilityLabelIconListTitle = '';
+
+  /** This id will be forwarded to the relevant inner element. */
+  @Prop() public iconListTitleId = `sbb-wagon-list-title-${++nextId}`;
+
+  /** Slotted Sbb-Icons. */
+  @State() private _icons: HTMLSbbIconElement[];
+
+  /** Host element. */
+  @Element() private _element!: HTMLElement;
+
+  public connectedCallback(): void {
+    this._readSlottedIcons();
+  }
+
+  /**
+   * Create an array with only the sbb-icon children.
+   */
+  private _readSlottedIcons(): void {
+    this._icons = Array.from(this._element.children).filter(
+      (e): e is HTMLSbbIconElement => e.tagName === 'SBB-ICON'
+    );
+  }
+
+  /**
+   * Create the accessibility text for the specific wagon types.
+   */
+  private _getAccessibilityText(): string {
+    if (this.type === 'wagon') {
+      let text = `${this.accessibilityLabelWagon} ${this.label}.`;
```

Comment:
I wonder if the separation of the accessibility properties actually provide an advantage? Could this not just be solved via one property?

Reply:
Switch to i18n